src/python/pants/backend/codegen/thrift/java/apache_thrift_java_gen.py

```diff
@@ -0,0 +1,60 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.backend.codegen.thrift.java.java_thrift_library import JavaThriftLibrary
+from pants.backend.codegen.thrift.java.thrift_defaults import ThriftDefaults
+from pants.backend.codegen.thrift.lib.apache_thrift_gen_base import ApacheThriftGenBase
+from pants.backend.jvm.targets.java_library import JavaLibrary
+from pants.base.exceptions import TargetDefinitionException
+from pants.binaries.thrift_binary import ThriftBinary
+
+
+class ApacheThriftJavaGen(ApacheThriftGenBase):
+  deprecated_options_scope = 'gen.thrift'  # New scope is gen.thrift-java.
+  deprecated_options_scope_removal_version = '1.5.0'
+
+  thrift_library_target_type = JavaThriftLibrary
+  thrift_generator = 'java'
+
+  _COMPILER = 'thrift'
+  _RPC_STYLE = 'sync'
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return (super(ApacheThriftJavaGen, cls).subsystem_dependencies() +
+            (ThriftDefaults, ThriftBinary.Factory.scoped(cls)))
+
+  @classmethod
+  def implementation_version(cls):
+    return super(ApacheThriftJavaGen, cls).implementation_version() + [('ApacheThriftGen', 2)]
```

Comment:
Well, this is just a string that gets mixed into the fingerprint, right? It can be anything. If I change it, it'll invalidate the world. Well, the world of thrift, anyway.  I guess that's not a big deal. Changed.